lib/Refactoring/Refactoring.cpp

```diff
@@ -8421,6 +8421,62 @@ bool RefactoringActionAddAsyncWrapper::performChange() {
   return false;
 }
 
+static MacroExpansionExpr *
+findMacroExpansionTargetExpr(const ResolvedCursorInfo &Info) {
+
+  // Handle '#' position in '#macroName(...)'.
+  if (auto exprInfo = dyn_cast<ResolvedExprStartCursorInfo>(&Info)) {
+    if (auto target =
+            dyn_cast_or_null<MacroExpansionExpr>(exprInfo->getTrailingExpr()))
+      if (target->getRewritten())
+        return target;
+    return nullptr;
+  }
+
+  // Handle 'macroName' position in '#macroName(...)'.
+  if (auto refInfo = dyn_cast<ResolvedValueRefCursorInfo>(&Info)) {
+    if (refInfo->isRef() && isa_and_nonnull<MacroDecl>(refInfo->getValueD())) {
+      ContextFinder Finder(
+          *Info.getSourceFile(), Info.getLoc(), [&](ASTNode N) {
+            auto *expr =
+                dyn_cast_or_null<MacroExpansionExpr>(N.dyn_cast<Expr *>());
+            return expr &&
+                   (expr->getMacroNameLoc().getBaseNameLoc() == Info.getLoc());
+          });
+      Finder.resolve();
+      if (!Finder.getContexts().empty()) {
+        auto *target =
+            dyn_cast<MacroExpansionExpr>(Finder.getContexts()[0].get<Expr *>());
+        if (target->getRewritten())
+          return target;
+      }
+    }
+    return nullptr;
+  }
+
+  // TODO: handle MacroExpansionDecl.
+  return nullptr;
```

Comment:
What's the TODO here? It's not like we can expand a decl directly.

Reply:
Macro expansion for `MacroExpansionDecl` in the type checker is not implemented yet.
We can implement refactoring for `MacroExpansionDecl`, but there's no way to test it.

Reply:
`MacroExpansionDecl` isn't implemented anywhere yet. When we get to declaration-producing macro expansions, we'll implement that support throughout.

Reply:
Yeah, I misread this as a `MacroDecl` :)